projects/hipblaslt/tensilelite/Tensile/Source/lib/source/analytical/StreamK.cpp

```diff
@@ -0,0 +1,395 @@
+                        << ", fixup_peers: " << fixup_peers_v2
+                        << ", cache_penalty: " << cache_penalty << ", m: " << m
+                        << ", n: " << n << ", k: " << k << ", a: " << a << ", b: " << b
+                        << ", c: " << c << ", d: " << d << std::endl;
+                    }
+
+                    if(min_grid_runtime.second > runtime)
+                    {
+                        min_grid_runtime.first  = g;
+                        min_grid_runtime.second = runtime;
+                    }
+
+                    if(min_grid_runtime_v2.second > runtime_v2)
+                    {
+                        min_grid_runtime_v2.first  = g;
+                        min_grid_runtime_v2.second = runtime_v2;
+                    }
+                }
+
+                if(verbose)
+                {
+                    std::cout << "[original] Number of Output Tiles: "
+                    << number_of_output_tiles(BLK_M, BLK_N, m, n, batch) << std::endl;
+                    std::cout << "[original] Minimum runtime: " << min_grid_runtime.second
+                    << " @ grid size: " << min_grid_runtime.first << std::endl;
+
+                    std::cout << "[cache-offset] Number of Output Tiles: "
+                    << number_of_output_tiles(BLK_M, BLK_N, m, n, batch) << std::endl;
+                    std::cout << "[cache-offset] Minimum runtime: " << min_grid_runtime_v2.second
+                    << " @ grid size: " << min_grid_runtime_v2.first << std::endl;
+                }
+
+                return min_grid_runtime_v2.first;
+            }
+
+            size_t select_streamk_grid(
+                size_t x,
+                size_t y,
+                size_t z,
+                size_t batch,
+                bool            trans_a,
+                bool            trans_b,
+                size_t          element_size_A,
+                size_t          element_size_B,
+                size_t          element_size_out,
+                DataType        mi_datatype,
+                size_t          workspace_size,
+                size_t          mt_m,
+                size_t          mt_n,
+                size_t          mt_k,
+                size_t          mi_m,
+                size_t          mi_n,
+                size_t          mi_k,
+                size_t          workgroup_mapping,
+                size_t          workspace_size_per_elem_c,
+                int             occupancy,
+                const Hardware& analytical_hardware,
+                int dynamic_grid_version)
+            {
+                int cu_count = analytical_hardware.N_CU;
+                int tiles = number_of_output_tiles(mt_m, mt_n, x, y, batch);
+
+                // Dynamically pick the minimum between the cu_count or number of tiles.
+                if(dynamic_grid_version == 1)
+                {
+                    return min(cu_count, tiles);
```

Comment:
Building in tritonBLAS:
```
projects/hipblaslt/tensilelite/Tensile/Source/lib/source/analytical/StreamK.cpp:261:28: error: ‘min’ was not declared in this scope; did you mean ‘std::min’?
        261 |                     return min(cu_count, tiles);
            |                            ^~~
            |                            std::min
      In file included from /usr/include/c++/11/functional:65,
                       from /home/muhosama/tritonBLAS/_origami/projects/hipblaslt/tensilelite/Tensile/Source/lib/include/Tensile/analytical/Utils.hpp:36,
                       from /home/muhosama/tritonBLAS/_origami/projects/hipblaslt/tensilelite/Tensile/Source/lib/source/analytical/StreamK.cpp:3:
      /usr/include/c++/11/bits/stl_algo.h:3455:5: note: ‘std::min’ declared here
       3455 |     min(initializer_list<_Tp> __l, _Compare __comp)
            |     ^~~
```

```suggestion
                    return std::min(cu_count, tiles);
```